src/main/scala/ayai/components/pathfinding/NeighboringPositionFinder.scala

```diff
@@ -0,0 +1,86 @@
+package ayai.components.pathfinding
+
+import ayai.components.Position
+
+trait NeighboringPositionFinder {
+  def neighboringPositions(pos: Position): List[Position]
+}
+
+/**
+ * Returns the positions marked 'x' on this diagram, where 'o' is the given position:
+ *
+ * .x.
+ * xox
+ * .x.
+ */
+class ManhattanNeighboringPositionFinder extends NeighboringPositionFinder {
+  override def neighboringPositions(pos: Position): List[Position] = {
+    val coordinateDifferences = List(
+      Tuple2(1, 0),
+      Tuple2(-1, 0),
+      Tuple2(0, 1),
+      Tuple2(0, -1)
+    )
+    coordinateDifferences.map(differences => differences match {
+      case Tuple2(xDiff, yDiff) => Position(pos.x + xDiff, pos.y + yDiff)
+    }).filterNot(position => ??? /* map(position) == NodeState.IMPASS */)
+    // Should this take a map or a graph?
+    // I do want to be able to look up the position just by a Position. That means it should define apply().
+    // But I can't use the NodeMatrixGraphView view for that, because that needs to call this to find neighbors.
+    // So the object with an apply() definition will need to be an intermediate value
+    // between an Array[Array[Node]] and a NodeMatrixGraphView
+    // I think I will make a NodeMatrix class, and make GraphFactory return it.
+    // There is one thing to check before I do: is it okay that we use `Position`s to index the grid?
```

Comment:
If I'm understanding you correctly, I _don't_ think we can address a Graph (i.e. one returned by `GraphFactory.generateGraph`, an `Array[Array[Node]]`) by `Position`.This function, which is present in `GraphFactory.scala` seems to suggest this:

```scala
private def convertPositionToGrid(position: Position, ratio: Float): Tuple2[Int, Int] =  {
    (math.round((position.x / ratio)), math.round((position.y / ratio)))
  }
```

That's used when setting a `(x, y)` coordinate when generating the graph to return. The giveaway there is the conversion from `Position` to the correct values using `ratio` which seems to be set from the `tileSize` of the map. If we _did_ want to use the a Graph like the one returned by `GraphFactory` (which I think you have here now), then we'll need to use a similar converter within `NodeMatrixGraphView` to convert from `Position` => `(Int, Int)` to address the matrix.

I think the graph is just a generalized version of the map that is not supposed to be dependent upon the tile size of the map. That said, I think it makes sense to use graph types for pathfinding. That way, pathfinding has no tie to tile size. In that case, we'd just pass around `Tuple2`s of positions which grid-based and convert them back to `Position`s at the end of the processing. What do you think?